idaes/generic_models/unit_models/surrogate_model.py

```diff
@@ -0,0 +1,135 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2020, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+Generic template for a surrogate unit model.
+"""
+
+from pyomo.environ import Reference, SolverFactory
+from pyomo.network import Port
+from pyomo.common.config import ConfigValue, In
+
+from idaes.core.process_base import declare_process_block_class, \
+    ProcessBlockData
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+
+__author__ = "Jaffer Ghouse"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("SurrogateModel")
+class SurrogateModelData(ProcessBlockData):
+    """
+    This is the class for a surrogate unit model. This will provide a user
+    with a generic skeleton to add a surrogate unit model.
+    """
+
+    # Create Class ConfigBlock
+    CONFIG = ProcessBlockData.CONFIG()
+    CONFIG.declare("dynamic", ConfigValue(
+        default=False,
+        domain=In([False]),
+        description="Dynamic model flag",
+        doc="""Surrogate model assumed to be steady-state."""))
+
+    def build(self):
+        """
+        General build method for SurrogateModelBlockData.
+
+        Inheriting models should call `super().build`.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        super(SurrogateModelData, self).build()
+
+    def add_ports(self, name=None, member_list=None, doc=None):
+        """
+        This is a method to build Port objects in a surrogate model and
+        populate this with appropriate port members as specified. User can add
+        as many inlet and outlet ports as required.
+
+        Keyword Args:
+            name : name to use for Port object.
+            dict : dictionary containing variables to be added to the port
```

Comment:
Is dict member_list?  If dict is member_list, them maybe member_dict would be more clear.